pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -276,3 +322,13 @@ func fitsRequest(podRequest *preFilterState, nodeInfo *framework.NodeInfo, ignor
 
 	return insufficientResources
 }
+
+// Score invoked at the Score extension point.
+func (f *Fit) Score(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	nodeInfo, err := f.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
```

Comment:
Do we want to register Fit as a score plugin in default provider? If so, we should check if its scoringStrategy is nil here.

Reply:
> Do we want to register Fit as a score plugin in default provider?

Yes, but registering is for a followup PR. 

> If so, we should check if its scoringStrategy is nil here.

Yes, I think it is better if we don't embed the `resourceAllocationScorer` in line 70 above, rather make it a reference that we check here (we could fatal since it shouldn't really happen if the plugin config is properly defaulted)

Reply:
Sort of. There is nothing being shared between the Filtering and Scoring parts, so I'm thinking if we should leave `Fit` as a pure Filter plugin, say not polymorphed to a Score plugin, and then put the scoring stuff into a separate Score struct.

Reply:
Other features, node affinity, taints/tolerations, pod affinity and topology spread all have filter and score, why not apply the same pattern here?

Reply:
By separating, I thought it might be possible to register it as a score plugin _dynamically_, and hence don't need to do the "if" check in Score() function. But it seems more complicated, so never mind.

Reply:
@ahg-g Maybe i'm not understand this `I think it is better if we don't embed the resourceAllocationScorer in line 70 above, rather make it a reference that we check here` clearly, but we do check scoringStrategy !=nil in newFit and ValidateNodeResourcesFitArgs.If fit as default score plugin, scoringStrategy == nil will not allowed?